cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,202 @@ def test_image_url_is_shown_on_movie_session_detail(self):
+            duration=20
+        )
+        movie_3 = sample_movie(
+            title="Movie",
+            description="description",
+            duration=30
+        )
+
+        actor_1 = sample_actor()
+        actor_2 = sample_actor(
+            first_name="Actor",
+            last_name="Test",
+        )
+
+        movie_2.actors.add(actor_2, )
+        movie_3.actors.add(actor_1, actor_2)
+
+        res = self.client.get(
+            MOVIE_URL,
+            {"actors": f"{actor_1.id},{actor_2.id}"}
+        )
+
+        filtered_movie = [movie_3, movie_2]
+        serializer = MovieListSerializer(filtered_movie, many=True)
+        serializer_without_actors = MovieListSerializer(movie_1)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            list(res.data), list(serializer.data)
+        )
+        self.assertNotIn(list(res.data), serializer_without_actors)
+
+    def test_filter_movie_by_genres(self):
+        movie_1 = sample_movie()
+        movie_2 = sample_movie(
+            title="test movie",
+            description="some test",
+            duration=20
+        )
+        movie_3 = sample_movie(
+            title="Movie",
+            description="description",
+            duration=30
+        )
+
+        genre_1 = sample_genre()
+        genre_2 = sample_genre(
+            name="Horror"
+        )
+
+        movie_2.genres.add(genre_2,)
+        movie_3.genres.add(genre_1, genre_2)
+
+        res = self.client.get(
+            MOVIE_URL,
+            {"genres": f"{genre_1.id},{genre_2.id}"}
+        )
+
+        filtered_movie = [movie_3, movie_2]
+        serializer = MovieListSerializer(filtered_movie, many=True)
+        serializer_without_genre = MovieListSerializer(movie_1)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            list(res.data), list(serializer.data)
+        )
+        self.assertNotIn(list(res.data), serializer_without_genre)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        movie.actors.add(sample_actor())
+        movie.genres.add(sample_genre())
+
+        res = self.client.get(detail_url(movie.id))
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        payload = {
+            "title": "new movie",
+            "description": "some",
+            "duration": 10
+        }
+        res = self.client.post(MOVIE_URL, payload)
+
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="admin@email.test",
+            password="adminpassword",
+            is_staff=True
+        )
```

Comment:
You can use here create_superuser method so 'is_staff' parameter won't be necessary to use here.